pkg/agent/externalnode/external_node_controller.go

```diff
@@ -0,0 +1,777 @@
+		ovsExternalIDHostIFName:               portData.ExternalIDs[ovsExternalIDHostIFName],
+		ovsExternalIDUplinkPort:               portData.ExternalIDs[ovsExternalIDUplinkPort],
+		ovsExternalIDEntityName:               eeName,
+		ovsExternalIDEntityNamespace:          portData.ExternalIDs[ovsExternalIDEntityNamespace],
+		ovsExternalIDIP:                       strings.Join(ips, eeSplitter),
+		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaHost,
+	}
+	err := c.ovsBridgeClient.SetPortExternalIDs(interfaceConfig.InterfaceName, attachInfo)
+	if err != nil {
+		return nil, err
+	}
+	ifIPs := make([]net.IP, 0)
+	for _, ip := range ips {
+		ifIPs = append(ifIPs, net.ParseIP(ip))
+	}
+	iface := &interfacestore.InterfaceConfig{
+		InterfaceName: interfaceConfig.InterfaceName,
+		Type:          interfacestore.ExternalEntityInterface,
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: interfaceConfig.PortUUID,
+			OFPort:   interfaceConfig.OFPort,
+		},
+		EntityInterfaceConfig: &interfacestore.EntityInterfaceConfig{
+			EntityName:      eeName,
+			EntityNamespace: interfaceConfig.EntityNamespace,
+			UplinkPort: &interfacestore.OVSPortConfig{
+				PortUUID: interfaceConfig.UplinkPort.PortUUID,
+				OFPort:   interfaceConfig.UplinkPort.OFPort,
+			},
+		},
+		IPs: ifIPs,
+	}
+	return iface, nil
+}
+
+func (c *ExternalNodeController) removeOVSPortsAndFlows(interfaceConfig *interfacestore.InterfaceConfig) error {
+	portUUID := interfaceConfig.PortUUID
+	portName := interfaceConfig.InterfaceName
+	if err := c.ofClient.UninstallVMUplinkFlows(portName); err != nil {
+		return err
+	}
+	port, err := c.ovsBridgeClient.GetPortData(portUUID, portName)
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			klog.ErrorS(err, "OVS port is not found", "port", portName)
+			return nil
+		}
+		return err
+	}
+	hostIfName := port.ExternalIDs[ovsExternalIDHostIFName]
+	uplinkIfName := port.ExternalIDs[ovsExternalIDUplinkName]
+	uplinkPortID := port.ExternalIDs[ovsExternalIDUplinkPort]
+	adapterConfig, er := util.GetUplinkConfig(hostIfName)
+	if er != nil {
+		klog.ErrorS(err, "Failed to get the configuration on the host interface", "hostInterface", hostIfName)
+		return err
+	}
+	if err := c.ovsBridgeClient.DeletePort(port.UUID); err != nil {
+		klog.ErrorS(err, "Failed to delete port in OVS", "hostInterface", hostIfName)
+		return err
+	}
+	if err := c.ovsBridgeClient.DeletePort(uplinkPortID); err != nil {
+		klog.ErrorS(err, "Failed to delete port in OVS", "uplinkIfName", uplinkIfName)
+		return err
+	}
+	defer func() {
+		// Delete host interface from OVS datapath if it is existing.
+		// This is to resolve an issue that OVS fails to remove the interface from datapath. It might happen because the interface
+		// is busy when OVS tries to remove it with the OVSDB interface deletion event.
+		if err := c.ovsctlClient.DeleteDPInterface(hostIfName); err != nil {
```

Comment:
I have synced with Anand and he has confirmed that hostIfName is also equal to the OVS internal port name for windows. As a result, I did not change this part. Meanwhile, I removed ovsExternalIDHostIFName from externalIDs since it is the same as InterfaceName in InterfaceConfig